llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -8131,15 +8131,69 @@ void SelectionDAGBuilder::visitIntrinsicCall(const CallInst &I,
 void SelectionDAGBuilder::visitConstrainedFPIntrinsic(
     const ConstrainedFPIntrinsic &FPI) {
   SDLoc sdl = getCurSDLoc();
+  const TargetLowering &TLI = DAG.getTargetLoweringInfo();
+  EVT VT = TLI.getValueType(DAG.getDataLayout(), FPI.getType());
+  fp::ExceptionBehavior EB = *FPI.getExceptionBehavior();
+  std::optional<RoundingMode> RM = FPI.getRoundingMode();
+
+  SDNodeFlags Flags;
+  if (EB == fp::ExceptionBehavior::ebIgnore)
+    Flags.setNoFPExcept(true);
+
+  if (auto *FPOp = dyn_cast<FPMathOperator>(&FPI))
+    Flags.copyFMF(*FPOp);
+
+  bool UseStaticRounding = EB == fp::ExceptionBehavior::ebIgnore && RM &&
+                           *RM != RoundingMode::Dynamic &&
+                           TLI.isStaticRoundingSupportedFor(FPI);
+  unsigned Opcode = 0;
+  SmallVector<SDValue, 4> Opers;
+  for (unsigned I = 0, E = FPI.getNonMetadataArgCount(); I != E; ++I)
+    Opers.push_back(getValue(FPI.getArgOperand(I)));
+
+  if (UseStaticRounding) {
+    switch (FPI.getIntrinsicID()) {
```

Comment:
It still doesn't make sense to me to select the constrained intrinsics to these operations. The metadata indicates that the runtime mode is this already, there isn't a benefit to using an instruction with a fixed mode

Reply:
Nodes like `STRICT_FADD` do not have information about rounding mode, they only prevent from undesired reordering by having side effect. After selector the information about rounding mode disappears.